crates/rattler_installs_packages/src/python_env/system_python.rs

```diff
@@ -11,13 +11,28 @@ pub enum FindPythonError {
 }
 
 /// Try to find the python executable in the current environment.
+/// Using sys.executable aproach will return original interpretator path
+/// and not the shim in case of using which
 pub fn system_python_executable() -> Result<PathBuf, FindPythonError> {
     // When installed with homebrew on macOS, the python3 executable is called `python3` instead
     // Also on some ubuntu installs this is the case
     // For windows it should just be python
-    which::which("python3")
-        .or_else(|_| which::which("python"))
-        .map_err(|_| FindPythonError::NotFound)
+    let output = std::process::Command::new("python3")
+        .arg("-c")
+        .arg("import sys; print(sys.executable, end='')")
+        .output()
+        .or_else(|_| {
+            std::process::Command::new("python")
+                .arg("-c")
+                .arg("import sys; print(sys.executable, end='')")
+                .output()
+        })
+        .map_err(|_| FindPythonError::NotFound)?;
```

Comment:
I'm guessing these could have multiple errors, maybe explicity match on `FileNotFound` and otherwise use a generic `io::Error`?

Reply:
Thank you very much for the suggestion!
I've added matching for subprocess errors and additional checking that sys.executable is not empty.